server/api/org/list/controller.js

```diff
@@ -0,0 +1,103 @@
+import Joi from 'joi';
+import compose from 'koa-compose';
+import last from 'lodash/last';
+import intersection from 'lodash/intersection';
+import packJSONRPC from '../../../middleware/packJSONRPC';
+import { validateRequest } from '../../../middleware/validation';
+import {
+  visitSession,
+  isUserAuthenticated,
+  visitUserPermissions,
+  getAuthorizedUniqueOrgIds,
+  findUserPermissionIndex,
+} from '../../../middleware/auth';
+import { AuthorizationError } from '../../../constants/errors';
+import getUserInfo from '../../user/info/service';
+import listOrgs, { parseCursor, stringifyCursor } from './service';
+
+const validationSchema = {
+  body: {
+    q: Joi.string()
+      .trim()
+      .max(2000)
+      .optional(),
+    limit: Joi.number()
+      .integer()
+      .min(1)
+      .max(500)
+      .default(100)
+      .optional(),
+    cursor: Joi.string()
+      .base64()
+      .optional(),
+    user: Joi.string()
+      .base64()
+      .optional(),
+  },
+};
+
+const isUserAuthorised = async ({ request }, next) => {
```

Comment:
The logic here is flawed. What you should do instead is...

1. request.body.user is specified
    1. Retrieve all orgs the designated user (request.body.user) is member of.
    2. Ensure requestor has the `yeep.org.read` permission in global scope (null) or at least 1 org or the user's orgs (as described above). Otherwise return error.
    3. Return orgs that are the intersection between requestor orgs and user orgs.
2. request.body.user is NOT specified
    1. Ensure requestor has the `yeep.org.read` permission in global scope (null) or at least 1 other org. Otherwise return error.
    2. Return orgs that are the requestor has access to (all orgs for `null` as described above).